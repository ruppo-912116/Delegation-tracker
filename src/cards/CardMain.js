import React, { Component } from 'react';
import Card1 from './cardsUI.js';
import Searchbar from "../search/Searchbar.js";
import {poolList, poolsUnite, totalPools, totalPoolsNumber} from "../services/poolList";
import {search} from "./utils";

// searchlist is to store the search result
// Search Variable is to store the search input
let SearchList;
let isMounted = true;
let Uppere = "";




export default class CardMain extends Component {

    state = {
        Pools : [],
        loading: false,
        error: false,
        errorMessage: "",
        SearchList:[]
    }

    async componentDidMount(){
        try{
        await totalPools()
        let index=  [];
        for(let i=1; i<= parseInt(totalPoolsNumber/100); i++){
         index.push(i)
        }
        await Promise.all(index.map(async i => {
            await poolList(i)
        }))
        this.setState({
        Pools: poolsUnite,
        loading: true,
        error:false,
        message: "Loading..."
        }, () => {
        if (poolsUnite && isMounted){
            this.setState({
                loading: false
            });
        }
        })
    }
    catch(err){
        this.setState({
            error: true,
            errorMessage: err.message,
            loading: false
        })
        console.log(err.message);
    }
    }

    // the function is for search method
    // upon search, this function is called and the state of the pools is changed
    searchTrigger = async (e) => {

        Uppere = e.toLowerCase();
        let resultPool = await search(`[BASE_URL_HERE]/pools?q=${Uppere}`)

        SearchList = resultPool

        this.setState({loading: false})

        if (SearchList.length === 0){
        this.setState({
        loading: true,
        message:"No pools found"
        })
        }   
    }


    render() {
        return (
            <div>
            {this.state.error ? <div className="d-flex justify-content-center">{this.state.errorMessage}</div>:
                <div id="mainId">
                <Searchbar trigger={this.searchTrigger}/>
                { this.state.loading ?
                 <div className="d-flex justify-content-center">{this.state.message}</div>   
                :<div>
                   {Uppere === "" ? <Card1 pools={this.state.Pools}/> : <Card1 pools={SearchList}/> }
                </div>
                }
                </div>
            }
            </div>
        )
    }
}

