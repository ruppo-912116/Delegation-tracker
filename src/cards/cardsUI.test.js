import { render, screen } from '@testing-library/react';
import Card1 from './cardsUI.js';
import {unmountComponentAtNode} from "react-dom";
import { act } from 'react-dom/test-utils';
import {BrowserRouter, MemoryRouter, Route, Switch } from "react-router-dom";
import { shallow, Enzyme } from "enzyme";
import renderer from 'react-test-renderer';
import {mount} from 'enzyme';
import CardMain from "./CardMain.js";


let container = null;
beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});


it ("should render correctly with no props", () => {
  const component = shallow(<Card1 />);
  expect (component).toMatchSnapshot();
});

it("should render correctly with given object", () => {
  const mockObject = {
    "name": "kfkerkfr"
  }

  const component = shallow(<Card1 pool={mockObject}/>);
  expect(component).toMatchSnapshot();
})



