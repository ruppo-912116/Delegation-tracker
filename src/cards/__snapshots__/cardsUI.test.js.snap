// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly with given object 1`] = `ShallowWrapper {}`;

exports[`should render correctly with no props 1`] = `ShallowWrapper {}`;
